From 660ea522ae2bd0f5e5300aa52f936c346508470f Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 10 May 2017 12:12:35 +0300
Subject: [PATCH 0131/1239] arm64: mvebu: soc: a8k: add soc early init

SoC early init is required in order to print the
SoC information at the beginning of bootflow.

The added part will be used for the
sample-at-reset register init.

Change-Id: I5728a8493dd35835174ce41a1bf71c32956bbc7d
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
---
 arch/arm/mach-mvebu/armada8k/soc.c    | 11 +++++++++++
 board/Marvell/mvebu_armada-8k/board.c |  9 ++++++++-
 2 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-mvebu/armada8k/soc.c b/arch/arm/mach-mvebu/armada8k/soc.c
index 586e844043..933eb2ba49 100644
--- a/arch/arm/mach-mvebu/armada8k/soc.c
+++ b/arch/arm/mach-mvebu/armada8k/soc.c
@@ -8,6 +8,7 @@
 #include <common.h>
 #include <asm/io.h>
 #include <asm/arch/soc.h>
+#include <mvebu/mvebu_chip_sar.h>
 
 #define CP_DEV_ID_STATUS_REG		(MVEBU_REGISTER(0x2400240))
 #define DEVICE_ID_STATUS_MASK		0xffff
@@ -142,6 +143,16 @@ void soc_print_device_info(void)
 		printf("CP%x-A%d\n", cp_type, cp_rev);
 }
 
+int soc_early_init_f(void)
+{
+#ifdef CONFIG_MVEBU_SAR
+	/* Sample at reset register init */
+	mvebu_sar_init();
+#endif
+
+	return 0;
+}
+
 #ifdef CONFIG_ARCH_MISC_INIT
 int arch_misc_init(void)
 {
diff --git a/board/Marvell/mvebu_armada-8k/board.c b/board/Marvell/mvebu_armada-8k/board.c
index 30a6fb55ca..aa6b9126f1 100644
--- a/board/Marvell/mvebu_armada-8k/board.c
+++ b/board/Marvell/mvebu_armada-8k/board.c
@@ -16,9 +16,16 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+int __soc_early_init_f(void)
+{
+	return 0;
+}
+
+int soc_early_init_f(void) __attribute__((weak, alias("__soc_early_init_f")));
+
 int board_early_init_f(void)
 {
-	/* Nothing to do (yet), perhaps later some pin-muxing etc */
+	soc_early_init_f();
 
 	return 0;
 }
-- 
2.29.0

